andela/ah-the-answer-backend
# #164829367 Like/Dislike an Article

#### Description
This PR edits and adds functionality in the ‘articles’ app. These changes allow authenticated users to ‘like’ or ‘dislike’ an article. Changes to the ‘ArticleSerializer’ will now ensure that articles are fetched from the database with the number of likes and dislikes. The following endpoints were implemented as specified in the 'Users should be able to like and or dislike articles' feature:

|  Method  | Endpoint |  Description |
|---|---|---|
| POST  | articles/:slug/like/ | Likes a particular article |
| POST  | articles/:slug/dislike/ | Dislikes a particular article |

#### Description of Task to be completed?
An authenticated user should be able to like or dislike articles written by others as well as those crafted by the user

*NOTE*
this feature requires you to sign-up and login before you can like or dislike articles on the site.


#### How should this be manually tested?
```
git clone --single-branch --branch ft-like-dislike-article-164829367 https://github.com/andela/ah-the-answer-backend.git
```
After cloning the branch, move into the generated project folder;
```
cd ah-the-answer-backend
```
Setup environment;
```
virtualenv venv
source venv/bin/activate
```
Install dependencies;
```
pip install -r requirements.txt
```
Create .env file and ensure all the environmental variables are set
Deactivate virtual environment and activate it again;
```
deactivate
source .env
```
Make migrations;
```
python manage.py makemigrations
python manage.py migrate
```
Run the server;
```
python manage.py runserver
```
Send a GET request to retrieve all articles and have access to the slugs. If there are no articles create one via the endpoint;
```
 POST-http://127.0.0.1:8000/api/articles/
```
Pass in the JSON body with the article details and ensure there is an authenticated token in the header;
```
{
	"article": {
		"title": "The Red Moon",
		"description": "A monologue on the dying of the stars",
		"body": "Isn't it beautiful what the cosmos beholds, don't you feel gleeful and at awe when you cast your eyes at starry nights. The stellar age is but the pre-adolescence of the Universe, and the evolution of life on earth but a second in the great chase to the end of time. the sun will shrink, then grow red and fiery, and with its rise, the planets will burn"
	} 
}
``` 
Once it is created, the response body will in include a 'slug' field
Test the like/dislike article APIs using POSTMAN. the request should contain the bearer header with a valid token and the 'slug' for a particular article is passed through the URL, i.e. 
```
 http://127.0.0.1:8000/api/articles/the-red-moon-df9c1ec9da6f/like/
```
And body:
Run tests with the following command:
```
python manage.py test
```

#### What are the relevant pivotal tracker stories?
#164829376
